lib/WUI/nhttp/gcode_upload.cpp

```diff
@@ -144,41 +155,60 @@ GcodeUpload::UploadResult GcodeUpload::start(const RequestParser &parser, Upload
     fsync(fileno(file.get()));
     fseek(file.get(), 0, SEEK_SET);
 
-    char boundary_cstr[boundary.size() + 1];
-    memcpy(boundary_cstr, boundary.begin(), boundary.size());
-    boundary_cstr[boundary.size()] = '\0';
-    UploadState uploader(boundary_cstr);
+    if (parser.method == http::Method::Post){
 
-    if (const auto err = uploader.get_error(); get<0>(err) != Status::Ok) {
-        return StatusPage(get<0>(err), StatusPage::CloseHandling::ErrorClose, json_errors, get<1>(err));
-    }
+        const auto boundary = parser.boundary();
+        if (boundary.empty()) {
+            return StatusPage(Status::BadRequest, StatusPage::CloseHandling::ErrorClose, json_errors, "Missing boundary");
+        }
+
+        char boundary_cstr[boundary.size() + 1];
+        memcpy(boundary_cstr, boundary.begin(), boundary.size());
+        boundary_cstr[boundary.size()] = '\0';
+        UploadState uploader(boundary_cstr);
 
-    return GcodeUpload(move(uploader), json_errors, *parser.content_length, file_idx, move(file), uploaded);
+        if (const auto err = uploader.get_error(); get<0>(err) != Status::Ok) {
+            return StatusPage(get<0>(err), StatusPage::CloseHandling::ErrorClose, json_errors, get<1>(err));
+        }
+
+        return GcodeUpload(move(uploader), json_errors, *parser.content_length, file_idx, move(file), uploaded);
+    } else /*(parser.method == http::Method::Put)*/ {
```

Comment:
Is it? Will I get here even if I send another method? I didn't see anything checking for PUT anywhere on the path.